### ISSUE TITLE
Log rotation of kubelet producing garbage for crashlooping containers

### ISSUE THREAD

**Issue by loburm** [kubernetes/kubernetes#91765]
**What happened**:
I have written a simple container that generates a lot of logs (200KB/s) and crashes every two minutes. Exact numbers doesn't matter, the most important that container should generate at least 10MB of logs (default value), before it dies.

After checking /var/log/pods/{pod}/{container} directory of running container I see next picture:

> total 189M
>
> -rw-r--r-- 1 root root 6.9M Jun  4 08:03 1.log.20200604-080302.gz
> -rw-r--r-- 1 root root 6.5M Jun  4 08:04 1.log.20200604-080343.gz
> -rw-r--r-- 1 root root 6.3M Jun  4 08:05 1.log.20200604-080423.gz
> -rw-r----- 1 root root  12M Jun  4 08:05 1.log.20200604-080504
> -rw-r--r-- 1 root root 6.3M Jun  4 08:06 2.log.20200604-080614.gz
> -rw-r--r-- 1 root root 6.5M Jun  4 08:07 2.log.20200604-080655.gz
> -rw-r--r-- 1 root root 6.5M Jun  4 08:08 2.log.20200604-080736.gz
> -rw-r----- 1 root root  12M Jun  4 08:08 2.log.20200604-080816
> -rw-r--r-- 1 root root 6.9M Jun  4 08:10 3.log.20200604-080946.gz
> -rw-r--r-- 1 root root 6.5M Jun  4 08:11 3.log.20200604-081027.gz
> -rw-r--r-- 1 root root 6.5M Jun  4 08:11 3.log.20200604-081108.gz
> -rw-r----- 1 root root  12M Jun  4 08:11 3.log.20200604-081149
> -rw-r--r-- 1 root root 6.9M Jun  4 08:14 4.log.20200604-081329.gz
> -rw-r--r-- 1 root root 6.5M Jun  4 08:14 4.log.20200604-081410.gz
> -rw-r--r-- 1 root root 6.5M Jun  4 08:15 4.log.20200604-081451.gz
> -rw-r----- 1 root root  12M Jun  4 08:15 4.log.20200604-081531
> -rw-r--r-- 1 root root 5.8M Jun  4 08:18 5.log.20200604-081752.gz
> -rw-r--r-- 1 root root 6.5M Jun  4 08:19 5.log.20200604-081833.gz
> -rw-r--r-- 1 root root 6.5M Jun  4 08:19 5.log.20200604-081913.gz
> -rw-r----- 1 root root  12M Jun  4 08:19 5.log.20200604-081954
> -rw-r----- 1 root root 5.6M Jun  4 08:26 6.log
> -rw-r--r-- 1 root root 6.0M Jun  4 08:24 6.log.20200604-082345.gz
> -rw-r--r-- 1 root root 6.5M Jun  4 08:25 6.log.20200604-082425.gz
> -rw-r--r-- 1 root root 6.3M Jun  4 08:25 6.log.20200604-082506.gz
> -rw-r----- 1 root root  12M Jun  4 08:25 6.log.20200604-082546

Kubelet is not removing rotated files from previous run.

**What you expected to happen**:
Old rotated files should be removed. Otherwise single crashlooping container can take down the whole disk.

**How to reproduce it (as minimally and precisely as possible)**:
Deploy next pod on cluster with containerd running:
```
apiVersion: v1
kind: Pod
metadata:
  name: log-generator
spec:
  containers:
  - name: log-generator
    image: busybox
    args: ["sh", "-c", "i=1; while [ $i -le 100 ]; do j=1; while [ $j -le 2000 ]; do echo RandomStringRandomStringRandomStringRandomStringRandomStringRandomString; j=$(( j+1 )); done; sleep 1; i=$(( i+1 )); done"]
```

Determine node on which container is running:

`$ kubectl get pods -o wide | grep log-generator`

SSH into the node after few minutes and check present container logs:

`ls /var/log/pods/default_log-generator_4babe71d-a90c-4d77-89ec-8376d867864a/log-generator/ -lh`

**Environment**:
- Client Version: 

> version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.5-gke.0", GitCommit:"96891985a805dc058939eaddd98601988512add9", GitTreeState:"clean", BuildDate:"2020-04-16T23:59:48Z", GoVersion:"go1.13.9b4", Compiler:"gc", Platform:"linux/amd64"}

- GKE
- OS (e.g: `cat /etc/os-release`):
> BUILD_ID=12371.227.0
> NAME="Container-Optimized OS"
> KERNEL_COMMIT_ID=44db6154b00a43d87634fd0d35292aec1d0c14c7
> GOOGLE_CRASH_ID=Lakitu
> VERSION_ID=77
> BUG_REPORT_URL="https://cloud.google.com/container-optimized-os/docs/resources/support-policy#contact_us"
> PRETTY_NAME="Container-Optimized OS from Google"
> VERSION=77
> GOOGLE_METRICS_PRODUCT_ID=26
> HOME_URL="https://cloud.google.com/container-optimized-os/docs"
> ID=cos

- Kernel (e.g. `uname -a`): Linux gke-cluster-1-default-pool-717816a8-1mpm 4.19.112+ #1 SMP Sat Apr 4 06:26:23 PDT 2020 x86_64 Intel(R) Xeon(R) CPU @ 2.30GHz GenuineIntel GNU/Linux

Also I have seen the same bug on 1.15, so I believe that all versions of kubelet are affected by this issue. I don't see that it was fixed in 1.18 too: https://github.com/kubernetes/kubernetes/commits/master/pkg/kubelet/logs/container_log_manager.go

**Comment by tedyu** [comment]
Have you looked at #90334 ?
ref: #91211

**Comment by loburm** [comment]
It was fixed by #93333
